crates/cli/src/sarif/sarif_utils.rs

```diff
@@ -1279,91 +1296,175 @@ mod tests {
+                .into_iter()
+                .map(SarifRuleResult::try_from)
+                .collect::<Result<Vec<_>, _>>()
+                .map_err(anyhow::Error::msg)
+                .expect("getting results");
+
+            let sarif_report = generate_sarif_report(
+                &[rule.clone().into()],
+                &sarif_secret_results,
+                &"mydir".to_string(),
+                SarifReportMetadata {
+                    add_git_info: false,
+                    debug: false,
+                    config_digest: "5d7273dec32b80788b4d3eac46c866f0".to_string(),
+                    diff_aware_parameters: None,
+                    execution_time_secs: 42,
+                },
+            )
+            .expect("generate sarif report");
+
+            let expected_subset = serde_json::json!(
+            {
+              "runs": [
+                {
+                  "results": [
+                    {
+                      "fixes": [],
+                      "level": case.2,
+                      "locations": [
+                        {
+                          "physicalLocation": {
+                            "artifactLocation": {
+                              "uri": "myfile.py"
+                            },
+                            "region": {
+                              "endColumn": 2,
+                              "endLine": 2,
+                              "startColumn": 1,
+                              "startLine": 1
+                            }
+                          }
+                        }
+                      ],
+                      "message": {
+                        "text": "some secret"
+                      },
+                      "partialFingerprints": {},
+                      "properties": {
+                        "tags": [
+                          "DATADOG_CATEGORY:SECURITY",
+                          case.1,
+                        ]
+                      },
+                      "ruleId": "secret-rule",
+                      "ruleIndex": 0
+                    },
+                  ],
+                  "tool": {
+                    "driver": {
+                      "rules": [
+                        {
+                          "fullDescription": {
+                            "text": "secret-description"
+                          },
+                          "id": "secret-rule",
+                          "name": "secret-rule",
+                          "properties": {
+                            "tags": [
+                              "DATADOG_RULE_TYPE:SECRET"
+                            ]
+                          },
+                          "shortDescription": {
+                            "text": "secret-rule"
+                          }
+                        }
+                      ],
+                    }
+                  }
+                }
+              ],
+            }
+                    );
```

Comment:
Not a blocker, but I'm not sure if the indentation for the closing paren was expected (if it is for visual distinction from the json punctuation, that's totally fine)